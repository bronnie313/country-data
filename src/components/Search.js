import React from 'react';
import PropTypes from 'prop-types';
import './styles/Search.css';

const Search = ({ setSearchInput }) => {
  const handleSearchInput = (e) => {
    setSearchInput(e.target.value);
  };

  return (
    <div>
      <input
        className="search-input"
        type="search"
        name="search"
        onChange={handleSearchInput}
        id="search"
        placeholder="search counrty"
      />
    </div>
  );
};

Search.propTypes = {
  setSearchInput: PropTypes.string.isRequired,
};

export default Search;
