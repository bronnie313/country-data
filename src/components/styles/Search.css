.search-input {
  width: 13rem;
  padding: 4px;
  border: 1px solid gray;
  outline: none;
  border-radius: 3px;
}
