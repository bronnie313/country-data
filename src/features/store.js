import { configureStore } from '@reduxjs/toolkit';
import modalReducer from './modal/modalSlice';
import countryReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    modal: modalReducer,
    country: countryReducer,
  },
});

export default store;
