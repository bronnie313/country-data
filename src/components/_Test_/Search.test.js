import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../features/store';
import Search from '../Search';

describe('home component', () => {
  it('testing if page matches', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Search />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
