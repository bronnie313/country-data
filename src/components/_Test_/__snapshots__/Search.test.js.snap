// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component testing if page matches 1`] = `
<div>
  <input
    className="search-input"
    id="search"
    name="search"
    onChange={[Function]}
    placeholder="search counrty"
    type="search"
  />
</div>
`;
